### DISCUSSION TITLE
add enhancement to useOutsideClick to handle array of refs instead of one ref

### DISCUSSION THREAD

**Discussion by l3ardia** [chakra-ui/chakra-ui#9232] [Ideas] [answered]
Consider this situation:

You have a drop down menu and you need to close it when user clicks outside the menu, but there is one exception. You need to keep it open when user clicks on the menu icon, and it is placed outside the menu.

So you need array of refs to define as inside area including the menu and the menu icon.

**Comment by segunadebayo** [comment, marked as answer]
`useOutsideClick` is gone in v3. Use `useClickAway` from `react-use`. We are not shipping new v2 work.
